src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryCollectionExtension/CountryCollectionExtensionSpec.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Doctrine\QueryCollectionExtension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\QueryBuilder;
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+final class CountryCollectionExtensionSpec extends ObjectBehavior
+{
+    function let(UserContextInterface $userContext): void
+    {
+        $this->beConstructedWith($userContext);
+    }
+    function it_does_not_apply_conditions_for_admin(
+        UserContextInterface $userContext,
+        QueryBuilder $queryBuilder,
+        AdminUserInterface $admin,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        ChannelInterface $channel
+    ):void {
+        $queryBuilder->getRootAliases()->willReturn(['o']);
+
+        $userContext->getUser()->willReturn($admin);
+        $admin->getRoles()->willReturn(['ROLE_API_ACCESS']);
+
+        $queryBuilder->andWhere('o..id in :countries')->shouldNotBeCalled();
+        $queryBuilder->setParameter('countries', $channel->getCountries())->shouldNotBeCalled();
```

Comment:
In situation like this we could use argument helper to define, that any value will break this condition
```suggestion
        $queryBuilder->setParameter('countries', Argument::any())->shouldNotBeCalled();
```